frontend/src/components/App/Layout.tsx

```diff
@@ -44,11 +54,133 @@ const useStyle = makeStyles(theme => ({
+  const isSidebarOpen = useTypedSelector(state => state.ui.sidebar.isSidebarOpen);
+  const history = useHistory();
+
+  if (!cluster) {
+    return null;
+  }
+
+  if (!isClusterNotAvailable) {
+    return null;
+  }
+  return (
+    <>
+      <Box
+        display={'flex'}
+        justifyContent="center"
+        style={{
+          position: 'absolute',
+          color: theme.palette.common.white,
+          textAlign: 'center',
+        }}
+        bgcolor={'error.main'}
+        zIndex={1400}
+        width={'100%'}
+        p={0.5}
+        alignItems="center"
+      >
+        <Box ml={isSidebarOpen ? '-25vw' : 0}>
+          {t(`Something went wrong with cluster ${cluster}`)}
+        </Box>
+        <Box
+          style={{ cursor: 'pointer' }}
+          color={'error.main'}
+          bgcolor={theme.palette.common.white}
+          ml={1}
+          p={0.5}
+          onClick={() => {
+            history.replace('/');
+            changeClusterNotAvailableStateHandler(false);
+          }}
+        >
+          {t('frequent| Take me to cluster chooser')}
+        </Box>
+      </Box>
+    </>
+  );
+}
+
 export default function Layout({}: LayoutProps) {
   const classes = useStyle();
   const arePluginsLoaded = useTypedSelector(state => state.ui.pluginsLoaded);
-
+  const dispatch = useDispatch();
+  const clusters = useTypedSelector(state => state.config.clusters);
   const { t } = useTranslation('frequent');
+  const [isClusterNotAvailable, setIsClusterNotAvailable] = useState(false);
+
+  useEffect(() => {
+    window.clusterConfigFetchHandler = setInterval(
+      () => {
+        fetchConfig();
+      },
+      CLUSTER_FETCH_INTERVAL,
+      clusters
+    );
+    fetchConfig();
+    return () => {
+      if (window.clusterConfigFetchHandler) {
+        clearInterval(window.clusterConfigFetchHandler);
+      }
+    };
+  }, []);
+
+  const fetchConfig = () => {
+    const clusters = store.getState().config.clusters;
+    request('/config', {}, false, false)
+      .then((config: Config) => {
+        const clustersToConfig: ConfigState['clusters'] = {};
+        config?.clusters.forEach((cluster: Cluster) => {
+          clustersToConfig[cluster.name] = cluster;
+        });
+        const configToStore = {
+          ...config,
+          clusters: clustersToConfig,
+        };
+
+        if (clusters === null) {
+          dispatch(setConfig(configToStore));
+        } else if (Object.keys(clustersToConfig).length !== Object.keys(clusters).length) {
+          dispatch(setConfig(configToStore));
+        } else {
+          Object.keys(clustersToConfig).every((key: string) => {
+            const clustersCopy = _.cloneDeep(clusters[key]);
+            // remove the cluster status from the comparison
+            delete clustersCopy.useToken;
+            if (_.isEqual(clustersCopy, clustersToConfig[key])) {
+              return true;
+            } else {
+              dispatch(setConfig(configToStore));
```

Comment:
It seems this would dispatch multiple times if there are multiple changes, but we probably only want to dispatch once when multiple ones have changed?